expression/builtin_other.go

```diff
@@ -120,8 +191,22 @@ func (b *builtinInIntSig) evalInt(row chunk.Row) (int64, bool, error) {
 		return 0, isNull0, err
 	}
 	isUnsigned0 := mysql.HasUnsignedFlag(b.args[0].GetType().Flag)
-	var hasNull bool
-	for _, arg := range b.args[1:] {
+
+	args := b.args
+	if b.hashSet != nil {
```

Comment:
do we really need this optimization? access an empty map only costs ~2ns:
```go
func BenchmarkEmptyMapAccess(b *testing.B) {
	s := set.NewInt64Set()

	b.ResetTimer()
	for i := int64(0); i < int64(b.N); i++ {
		_, _ = s[i]
	}
}
```

benchmark result:
```sh
[jianzhang.zj:~/Code/tidb/expression] git:(master ✗)
➜ go test -bench BenchmarkEmptyMapAccess -run xx -count 5 -benchmem
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkEmptyMapAccess-12      542155310                1.98 ns/op            0 B/op          0 allocs/op
BenchmarkEmptyMapAccess-12      602839488                1.97 ns/op            0 B/op          0 allocs/op
BenchmarkEmptyMapAccess-12      602942157                1.98 ns/op            0 B/op          0 allocs/op
BenchmarkEmptyMapAccess-12      605327304                1.97 ns/op            0 B/op          0 allocs/op
BenchmarkEmptyMapAccess-12      603420452                1.95 ns/op            0 B/op          0 allocs/op
PASS
ok      github.com/pingcap/tidb/expression      7.452s
```

Reply:
we may need it in vectorized version?

Reply:
Yes, after rethinking, maybe we also need it in the non vectorized version. Suppose we have 1, 000, 000 row to process, this optimization saves 1 ms.